### PR TITLE
 ipstack integration

### PR DESCRIPTION
**Pull request by Arsenikr**
first version of ipstack integration, implementing !ip without dbot score due to license limitations

## Status
Ready

## Related Issues
https://github.com/demisto/etc/issues/15477

## Does it break backward compatibility?

   - No

## Must have
- [ x] Tests
- [ x] Documentation (in progress)
- [x] Code Review

### CODE REVIEW

File: Integrations/integration-Ipstack.yml [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        """
+        res = requests.request(
+            method=method,
+            url=BASE_URL + path
+        )
+
+        if not res.ok:
+            txt = 'error in URL {} status code: {} reason: {}'.format(url,res.status_code,res.text)
+            demisto.error(txt)
+            raise exception(txt)
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]:
```suggestion
            raise Exception(txt)
```

File: Integrations/integration-Ipstack.yml [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+
+        try:
+            res_json = res.json()
+            if res_json.get('code'):
+                txt = 'error in URL {} status code: {} reason: {}'.format(url,res.status_code,res.text)
+                demisto.error(txt)
+                raise exception(txt)
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]:
```suggestion
                raise Exception(txt)
```

File: Integrations/integration-Ipstack.yml [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+            else:
+                return res_json
+
+        except Exception as ex:
+            demisto.debug(str(ex))
+            demisto.results( { "Type" : entryTypes["error"], "ContentsFormat" : formats["text"], "Contents" : res.text } )
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]: no need to catch the exception here, you can it in upper level

File: Integrations/integration-Ipstack.yml [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+
+    ''' Commands '''
+
+    def do_ip(ip):
+
+        demisto.log(ip)
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]: remove demisto.log from the code, for debug use demisto.debug

File: Integrations/integration-Ipstack.yml [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+                    'Location': "{},{}".format(raw_response.get('latitude'),raw_response.get('longitude')),
+                    'Country': raw_response.get('country_name')
+                }
+            },
+            'Ipstack.ip(val.ID==obj.ID)': {
+                'address': raw_response.get('ip'),
+                'type': raw_response.get('type'),
+                'continent_name':raw_response.get('continent_name') ,
+                'latitude': raw_response.get('latitude'),
+                'longitude': raw_response.get('longitude'),
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]: country_name missing here

File: Integrations/integration-Ipstack.yml [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    - contextPath: Ipstack.IP.type
+      description: ipv4\ipv6
+      type: string
+    - contextPath: Ipstack.IP.continent_name
+      description: Continent of the IP
+      type: string
+    - contextPath: Ipstack.IP.latitude
+      description: Latitude
+      type: string
+    - contextPath: Ipstack.IP.longitude
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]: add country_name here too

File: Integrations/integration-Ipstack.yml [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+  defaultvalue: "true"
+  type: 8
+  required: false
+script:
+  script: |-
+    import requests
+    import json
+
+    BASE_URL = 'http://api.ipstack.com'
+    API_KEY = demisto.params().get('apikey')
```
Comment (anara123) [inline Integrations/integration-Ipstack.yml]: support also insecure param. see other integrations

**Comment by kirbles19** [comment]
Hi Arseny,

please add me as an assignee, and remove as reviewer

Get Outlook for Android<https://aka.ms/ghei36>

________________________________

File: Integrations/Ipstack/Ipstack.py [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+    except Exception as ex:
+        demisto.debug(str(ex))
+        demisto.results({"Type": entryTypes["error"], "ContentsFormat": formats["text"], "Contents": res.text})
+
+
+''' Commands '''
+
+
+def do_ip(ip):
+    demisto.log(ip)
```
Comment (anara123) [inline Integrations/Ipstack/Ipstack.py]: remove demisto.log

File: Integrations/Ipstack/Ipstack_description.md [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+## Ipstack Playbook
+---
+
+## Use Cases
+---
+  * Get Location info about an IP
+## Configure Ipstack on Demisto
+---
+
+1. Navigate to __Settings__ > __Integrations__ > __Servers & Services__.
```
Comment (anara123) [inline Integrations/Ipstack/Ipstack_description.md]: detailed description should contain only the necessary info to config the instance.
how to create API Key, permissions required, etc.
not the whole documentation generated. This documentation should be attached to new Documentation issue

**Comment by kirbles19** [comment]
All done.

**Comment by anara123** [comment]
@Arsenikr resolve the conflict

**Comment by yardensade** [comment]
- Why yml's description field has "\n\n" in the end? it breaks the generated RN.
- Why description.md contains entire documentation and not just the integration's description. (I can see it was commented in the code review and marked as "resolved" even tho it was not)

Please fix this before the next content freeze.

**Pull request merged**